### ISSUE TITLE
Trainer effective_batch_size property

### ISSUE THREAD

**Issue by lendle** [Lightning-AI/lightning#16083]
### Description & Motivation

It's [recommended](https://pytorch-lightning.readthedocs.io/en/latest/accelerators/gpu_faq.html#how-should-i-adjust-the-learning-rate-when-using-multiple-devices) to scale learning rate by effective batch size. Per the docs, effective batch size depends on strategy. It would be helpful to have this calculated if/when possible. 

### Pitch

Add an `effective_batch_size` property to `Trainer` based on the strategy, `batch_size * num_devices * num_nodes` for most strategies, just `batch_size` for DP.

### Alternatives

If it's not possible to get batch size (may not be for IterableDataLoader? I haven't used it personally), then add a property that computes a batch size scaling factor as a function of strategy, or throw a useful error.
